web/src/NetworkManager.jsx

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright (c) [2022] SUSE LLC
+ *
+ * All Rights Reserved.
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of version 2 of the GNU General Public License as published
+ * by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, contact SUSE LLC.
+ *
+ * To contact SUSE LLC about this file by physical or electronic mail, you may
+ * find current contact information at www.suse.com.
+ */
+
+import React, { useEffect, useReducer, useState } from "react";
+import { useInstallerClient } from "./context/installer";
+import Popup from "./Popup";
+import { Button, Text } from "@patternfly/react-core";
+
+const initIpData = {
+  addresses: [],
+  hostname: ""
+};
+
+const reducer = (state, action) => {
+  const data = action.payload;
+
+  switch (action.type) {
+    case "READ": {
+      return {
+        ...state,
+        ...data
+      };
+    }
+    default: {
+      return state;
+    }
+  }
+};
+
+function formatIp(address, prefix) {
+  return address + "/" + prefix;
+}
+
+function LoadNmConfig() {
+  const client = useInstallerClient();
+  const [state, dispatch] = useReducer(reducer, initIpData);
+
+  useEffect(() => {
+    const config = async () => {
+      const data = await client.network.config();
+
+      dispatch({
+        type: "READ",
+        payload: data
+      });
+    };
+
+    config();
+  }, [client.network]);
+
+  return state;
+}
+
+export default function DetailsPopup() {
+  const [isOpen, setIsOpen] = useState(false);
+
+  const open = () => setIsOpen(true);
+  const close = () => setIsOpen(false);
+
+  const state = LoadNmConfig();
+  const ips = state.addresses.map((addr) => formatIp(addr.address, addr.prefix));
+  const firstIp = ips.length > 0 ? ips[0] : "";
+
+  return (
+    <>
+      <Button variant="link" onClick={open}>
+        { firstIp } ({state.hostname})
+      </Button>
+
+      <Popup
+        isOpen={isOpen}
+        title={state.hostname}
+        aria-label="IP Addresses"
+      >
+        <Text>
+          { ips.map((ip) => <p key={ip}> {ip} </p>) }
```

Comment:
Note: Patternfly's `<Text>` component  is already _tranlated_ to an HTML `<p>` element by default. See https://www.patternfly.org/v4/components/text#text

Thus, I'd use 

```jsx
{ ips.map(ip => <Text key={ip}>{ip}</Text>) }
``` 

Although maybe it's better to use a plain [`<List>`](https://www.patternfly.org/v4/components/list#plain). No strong opinion at this moment.

Reply:
NP: as per [HTML specification](https://html.spec.whatwg.org/#the-p-element), `<p>` element is categorized as [_flow content_](https://html.spec.whatwg.org/#flow-content-2) and [_palpable content_](https://html.spec.whatwg.org/#palpable-content-2) and can only contain [_phrasing content_](https://html.spec.whatwg.org/#phrasing-content-2). A bit messy way to say that a `<p>` must not contain another `<p>`.

Reply:
So, React, d-bus and now patternfly ... I'm like Columbus in India ;-)